### DISCUSSION TITLE
Rust Gamedev Showcase

### DISCUSSION THREAD

**Discussion by AngelOnFira** [rust-gamedev/wg#121] [Ideas]
I would really like to get footage of games made in Rust so that a montage could be made to show off what is possible in the current ecosystem.

I've set up a basic form with questions that should work, does anyone have any thoughts about them? I also put very little work into the last one about the legal stuff; that might not fly for what we want. I can probably find examples of how other engines like Godot do it maybe? Not sure.

https://forms.gle/QsdSJk8Eoj1ng8Xc8

**Comment by AngelOnFira** [comment]
Also, I wonder if instead any added footage should be accessible to anyone as raw footage? I don't think I need to have it only under my lock and key.

**Reply by erlend-sh** [reply]
Maybe so, but 99% of people will only bother checking out the edited video anyhow.

More importantly, I highly recommend setting more specific requirements for footage, e.g. one file format and min/max sizes. Otherwise the job of the video editor will be far more laborious.

**Reply by nxcco** [reply]
I also think so. There should be requirements to the footage. Maybe some developers can even come up with some sort of a 1-2min "technical trailer", where they show the game and outline what the journey was and is like for them with Rust.

**Comment by Lokathor** [comment]
This would mostly be a way to show off art assets I think.

How good something looks in a showcase reel doesn't have much to do with the programming language or framework or basically anything that a programmer does.

**Reply by AngelOnFira** [reply]
I'm not sure I agree with this. To someone on the outside, a showcase doesn't only show that game development is possible with a particular engine/language, but also that it's _actually_ being done. Rust is still known to many people as a language that is still "immature" without enough of an ecosystem to put something cohesive together. Or maybe people might think that Rust's paradigm doesn't lend itself well to game development. Several of my friends have been surprised when I told them that I work on a game in Rust. To them it was just a systems language that could only solve the black-box of "systems problems" (to be fair; game dev does fit in here, but systems programming can be a loaded term).

I definitely think that the [Godot showcase](https://www.youtube.com/watch?v=iAceTF0yE7I) is a good example of this. For people that try to say that "Godot isn't as good as \<other engine\>; I haven't seen any real Godot games", a showcase can be an easy answer to that.

Finally, and maybe most importantly, a showcase is also a platform for devs to show off work they're proud of. I definitely see [the games section of arewegameyet](https://arewegameyet.rs/#games) promoting this as well. And to that extent, the art assets are indeed important!

That all being said, I can agree with your point from the extent that someone with years of industry game development experience might not gain anything from something like this. But I do think it can do good for the Rust game dev ecosystem :smiley: 

Interested to hear your thoughts on this!

**Reply by alice-i-cecile** [reply]
I agree with @AngelOnFira here: one of the big complaints I hear is "Bevy (and Rust) are not ready for serious projects yet; all I've seen are crude toys". The existence of a polished game shows the *possibility* to polish a game: it requires small things like audio, animation, windowing, input handling, UI and so on to all come together nicely.

Effectively what this sort of initiative provides is "social proof": it's a bit silly, but folks who aren't early adopters want to see that others have already invested in something, and are finding success. It gives them the confidence to investigate something more fully, and see for themselves how much they like the programming language and frameworks.

**Comment by alice-i-cecile** [comment]
Feedback on the questions:

1. "What is the description of your project" -> "Briefly describe your project"
2. "For example YouTube, Google Drive, or any other file storage platform. Ideally uncompressed video would best." -> "For example YouTube, Google Drive, or another file storage platform. Uncompressed video would best, as we are intending to create a montage reel and want to show your game at its best. Please make sure to provide game audio if any."
3. "Provide any other relevent links to the project" -> "When showcasing your game, which link would you like us to point viewers to so they can learn more?"
4. "I agree to let the Rust Gamedev WG use this footage in whatever capacity they like" -> "I agree to let the Rust Gamedev WG take segments of this footage to produce a highlight reel demonstrating Rust's potential for game development, as long as they provide appropriate attribution." Much fairer and more tightly scoped.
5. I think we should have a drop-down box for the stage of the project: "Pre-Alpha / Alpha / Beta / Early Access / Released". This is useful for viewers trying to evaluate things as both players and industry professionals.

**Reply by AngelOnFira** [reply]
Thanks! I've updated the form :)

**Comment by aclysma** [comment]
I think this is a good idea, not because we need to "prove" anything on a technical level (people ship real, commercially successful games on all kinds of tech stacks of.. uh.. varying quality), but because I think people in the community would enjoy participating in it. It may, as a side effect, show that rust gamedev is a "happening" place with a community of active people. But I think the best reason for doing it is because many people in our community would enjoy participating in it and seeing the result. :)

(Also I don't know what to do about the last question.. but that seems like something we can bikeshed later.)

As far as what the WG role in this would be, I think it would just be getting the word out?

**Reply by AngelOnFira** [reply]
Just touching on the WG role for this, I think it makes sense for this to be a WG initiative rather than just a "me" initiative. It aligns pretty well with the general goals of the WG (showing off what the community is doing). But let me know if there are other items to consider :)

**Comment by AngelOnFira** [comment]
I'm looking to get this sent out soon, can I get one last check on it? I'll probably post it in a few hours:

- Rust Twitter
- /r/rust_gamedev
- Game Dev in Rust Discord server
- Rust Discord #gamedev-wg channel
- Let me know if I should post anywhere else!

[Form Link](https://forms.gle/hXaUDHkerxKkUZC16)

**Reply by kvark** [reply]
Do I need to re-fill it if I already submitted my answers on an earlier version?

**Reply by AngelOnFira** [reply]
No need, I'll just DM you if I need anything else :)

**Comment by voxelias** [comment]
Great Idea!
For how long the form will be available? Would be nice to have that info somewhere. Or I am blind and don't see it?

We don't have any showcase video yet, maybe we can prepare some.  Depends on how much time we have...

**Reply by AngelOnFira** [reply]
I don't have a particular deadline, but I agree it would help move people along. Maybe I'll set this up as the "2022 Showcase" and add a deadline of around 2 weeks. Great point!

**Reply by AngelOnFira** [reply]
I've added this

> This form will close on the 12th of March (won't be finalized until the 14th to accommodate timezones), and a "2022 Showcase" will be created.
